SPEC.md

```diff
@@ -307,6 +307,8 @@ For invoking a callback URL:
 - Include any callback headers supplied in the originating StartOperation request, stripping away the `Nexus-Callback-`
   prefix.
 - Include the `Nexus-Operation-State` header.
+- Include the `Nexus-Operation-Id` header and any `Nexus-Link` headers for resources associated with this operation to
+  support completing asynchronous operations before the response to StartOperation is received.
```

Comment:
I assume when we allow users to have external callbacks this won't expose any security risks by giving the callback internal information like a workflow that may be backing an operation? Or at that time we will have some way to hide the workflow link if a Nexus endpoint doesn't want to expose it?

Reply:
Yes, there needs to be a knob for users to decide if they want to expose caller and handler information in Temporal. For now, we're limited to same account traffic and the consensus seems to be to always expose.

Reply:
We should also include the `Nexus-Operation-Start-Time` header to be able to recreate the full started event.